### DISCUSSION TITLE
How to make a node/deno compatible ts module - import statements

### DISCUSSION THREAD

**Discussion by firstdorsal** [denoland/deno#13470] [Q&A]
first of all: deno is great!

I want to create a typescript codebase that works with deno but also compiles to js to be executed with nodejs

in ts the import has to end in js else node wont know where to load the files from
```ts
import { vaultLoginUserpass, getVaultValue } from "./vault/vault.js";
```
but in deno the files wont load from js because the path is (of course) plain wrong.
I know this is typescripts incompetence but is there a nice way in deno to fix this?
i tried: https://deno.land/manual@v1.1.0/getting_started/typescript#compiler-hint
but this is neither very comfortable nor seems to work

thank you

**Comment by RemyRylan** [comment]
For my projects I use the [built-in bundler](https://deno.land/manual/tools/bundler) to produce a dependency-free JS file. As long as you don't use the `Deno` namespace in your code, it should work.
